src/pools/abstract-pool.ts

```diff
@@ -0,0 +1,172 @@
+import EventEmitter from 'events'
+import type { MessageValue } from '../utility-types'
+
+export type ErrorHandler<Worker> = (this: Worker, e: Error) => void
+export type OnlineHandler<Worker> = (this: Worker) => void
+export type ExitHandler<Worker> = (this: Worker, code: number) => void
+
+export interface PoolOptions<Worker> {
```

Comment:
We could consider to have more specific options since that worker_thread has some parameters that cluster does not and viceversa.

Is better to keep options specific

Reply:
Yeah it's possible to define types **if needed** later on like 
```ts
export interface FixedThreadPoolOptions extends PoolOptions<ThreadWorkerWithMessageChannel> {
  additionalProperty: boolean;
  // ...
}

Reply:
Sounds good it should work for later, I am not familiar with Typescript but should work fine

Reply:
Example can now be seen via `ClusterPoolOptions` in commit https://github.com/pioardi/poolifier/pull/116/commits/6e19fd2e0a86ec4e3cc78b70cadc76da1e086876